sklearn/svm/tests/test_svm.py

```diff
@@ -422,24 +422,154 @@ def test_weight():
         assert f1_score(y_[100:], y_pred) > .3
 
 
-def test_sample_weights():
-    # Test weights on individual samples
-    # TODO: check on NuSVR, OneClass, etc.
-    clf = svm.SVC()
-    clf.fit(X, Y)
-    assert_array_equal(clf.predict([X[2]]), [1.])
+@pytest.mark.parametrize("estimator", [svm.SVC(C=1e-2), svm.NuSVC()])
```

Comment:
This test doesn't check zero or negative weights. I added it to the unit tests for  this feature as asked by @glemaitre who was working on another feature related to  weights in SVM. Probably that feature is already merged.

Reply:
If you check the previous `test_sample_weights`, it was not testing anything from the `sample_weight`.
The added tests do not fail because it was no bug because at least they are added to be sure that the results are rights. It could be done in another PR but I wanted to avoid splitting it for simplicity for a first contrib :)